embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/SystemInfo.kt

```diff
@@ -0,0 +1,56 @@
+package io.embrace.android.embracesdk.internal
+
+import android.os.Build
+import io.opentelemetry.semconv.incubating.OsIncubatingAttributes
+
+/**
+ * Information about the the device or OS that can be retrieved without disk or platform API access
+ */
+internal data class SystemInfo(
+    val osName: String = "android",
+    val osType: String = OsIncubatingAttributes.OsTypeValues.LINUX,
+    val osBuild: String = getOsBuild(),
+    val osVersion: String = getOsVersion(),
+    val osVersionName: String = getOsVersionName(),
+    val deviceManufacturer: String = getDeviceManufacturer(),
+    val deviceModel: String = getDeviceModel()
+)
+
+internal fun getOsVersion(): String {
```

Comment:
Anything that accesses `Build` will throw in tests without the right test runner to fake the Android APIs. Use these wrapping functions so it doesn't happen.